src/cmd/services/m3query/config/config.go

```diff
@@ -228,6 +231,46 @@ func (c QueryConfiguration) TimeoutOrDefault() time.Duration {
 	return defaultQueryTimeout
 }
 
+// RestrictTagsAsStorageRestrictByTag returns restrict tags as
+// storage options to restrict all queries by default.
+func (c QueryConfiguration) RestrictTagsAsStorageRestrictByTag() (*storage.RestrictByTag, bool, error) {
+	if c.RestrictTags == nil {
+		return nil, false, nil
+	}
+
+	var (
+		cfg    = *c.RestrictTags
+		result = handleroptions.StringTagOptions{
+			Restrict: make([]handleroptions.StringMatch, 0, len(cfg.Restrict)),
+			Strip:    cfg.Strip,
+		}
+	)
+	for _, elem := range cfg.Restrict {
+		value := handleroptions.StringMatch(elem)
+		result.Restrict = append(result.Restrict, value)
+	}
+
+	opts, err := result.StorageOptions()
+	if err != nil {
+		return nil, false, err
+	}
+
+	return opts, true, nil
+}
+
+// RestrictTagsConfiguration applies tag restriction to all queries.
+type RestrictTagsConfiguration struct {
+	Restrict []StringMatch `yaml:"match"`
+	Strip    []string      `yaml:"strip"`
+}
+
+// StringMatch is an easy to use representation of models.Matcher.
+type StringMatch struct {
+	Name  string `yaml:"name"`
+	Type  string `yaml:"type"`
+	Value string `yaml:"value"`
+}
```

Comment:
This can probably reuse this one: https://github.com/m3db/m3/blob/master/src/query/api/v1/handler/prometheus/handleroptions/tag_options.go#L101

Reply:
I am avoiding reusing it since I want to set the "yaml" fields =[